cql3/statements/ks_prop_defs.cc

```diff
@@ -187,12 +190,11 @@ static locator::replication_strategy_config_options prepare_options(
             throw exceptions::configuration_exception(fmt::format(
                     "Rack list for '{}' contains duplicate entries", dc));
         }
-        if (old_rf && !old_rf->is_rack_based() && old_rf->count() != 0) {
-            // FIXME: Allow this if replicas already conform to the given rack list.
-            // FIXME: Implement automatic colocation to allow transition to rack list.
-            throw exceptions::configuration_exception(fmt::format(
-                    "Cannot change replication factor from numeric to rack list for '{}'", dc));
-        }
+        numeric_to_rack_list_transition = numeric_to_rack_list_transition || (old_rf && !old_rf->is_rack_based() && old_rf->count() != 0);
+    }
+
+    if (numeric_to_rack_list_transition && rf_change) {
+        throw exceptions::configuration_exception("Cannot change replication factor from numeric to rack list and rf value at the same time");
```

Comment:
But this is allowed:
'dc1': ['rack1'] -> ['rack1', 'rack2']
'dc1: 1 -> ['rack1']

Should we validate per-DC?

Reply:
It isn't allowed. `keyspace_rf_change` is per cluster, not per DC and we process them all together.

If we want to allow the above then we can as well allow both changes in the same DC.

Reply:
Right